### PR TITLE
Fixed several bugs and added additional debugging information.

### PR DESCRIPTION
**Pull request by IharBury**
- Additional debugging information added to debug builds.
- Fixed a bug in LiquidProjections.NEventStore which could lead to projectors hanging when getting the first transaction from the cache
- Fixed a bug in LiquidProjections.NEventStore which could lead to projectors hanging when the cache access throws an exception
- Fixed a bug in NEventStoreAdapter which could lead to unexpected sleep times when the requested page size is bigger than the NEventStore internal max page size
- Using checkpoint 0 to represent the state before the first transaction instead of null internally
- Renamed several checkpoint parameters/fields/properties to clarify that they should contain the previous checkpoint (instead of the checkpoint being loaded)

### CODE REVIEW

**Comment by dennisdoomen** [comment]
So what was really causing the hang?

**Comment by IharBury** [comment]
@dennisdoomen:

>So what was really causing the hang?

The major problem was Page.PreviousCheckpoint having non-equal values `null` and `0` representing the beginning of the history. That led to an infinite loop while successfully getting the transactions from the cache inside the loader task.

**Comment by dennisdoomen** [comment]
Can we create a spec that shows we had this problem? Like the one in the other PR?

**Comment by IharBury** [comment]
@dennisdoomen:

> Can we create a spec that shows we had this problem? Like the one in the other PR?

Done

File: Src/LiquidProjections/IEventStoreWithSubscriptionIds.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LiquidProjections
+{
+    public interface IEventStoreWithSubscriptionIds : IEventStore
+    {
+        IDisposable Subscribe(long? lastProcessedCheckpoint, Func<IReadOnlyList<Transaction>, Task> handler,
```
Comment (dennisdoomen) [inline Src/LiquidProjections/IEventStoreWithSubscriptionIds.cs]: I suspect you introduced this extra interface to ensure backwards-compatibility?

Comment (IharBury) [inline Src/LiquidProjections/IEventStoreWithSubscriptionIds.cs]: Exactly

**Pull request closed without merging**